public/app/features/dashboard-scene/pages/DashboardScenePageStateManager.ts

```diff
@@ -290,13 +303,17 @@ export class DashboardScenePageStateManager extends StateManagerBase<DashboardSc
   }
 
   public getCacheKey(cacheKey: string): string {
-    const scopesCacheKey = getSelectedScopesNames().sort().join('__scp__');
+    // The following are the scopes
+    // We need those as there is a feature where we pass the scopes to fetch dashboard API hence we need to cache based on these
+    const scopes = getScopes(true);
+    const scopesCacheKey = scopes ? `__scp__${scopes.join(',')}` : '';
 
-    if (!scopesCacheKey) {
-      return cacheKey;
-    }
+    // The following are the query params for time range and filters
+    // We need those as there is a feature where we pass these query params to fetch dashboard API hence we need to cache based on these
+    const queryParams = getTimeRangeAndFilters(true);
+    const queryParamsKey = queryParams ? `__qp__${urlUtil.serializeParams(queryParams)}` : '';
 
-    return `${cacheKey}__scp__${scopesCacheKey}`;
+    return `${cacheKey}${scopesCacheKey}${queryParamsKey}`;
```

Comment:
why do we need to change the cache key? we always check the version of the dashboard cached and the one returned by the API, so as long as they return different versions of the dashboard we only need to check that

Reply:
Removed the additional cache key params.